classify-cluster-priority/README.md

```diff
@@ -0,0 +1,85 @@
+# Classifying clusters based on priority
+
+Contents:
+
+1. User stories
+2. Current practice
+3. Proposal
+4. Considerations and explanations
+5. Possible applications
+
+## User stories
+
+- As a cluster admin, I want to easily communicate the purpose and importance of a cluster to those interacting with it, so that they can act accordingly.
+- As Giant Swarm operations staff I want to understand the importance of clusters so I can make appropriate decisions, e. g. when prioritizing problem mitigation in an incident affecting multiple clusters.
+
+## Current practice
+
+- One customer uses the first character of the cluster name to distinguish between production (`p`), development (`d`), and staging (`s`). This customer also has clusters which don't fall into either of these categories and their names start with arbitrary letters.
+- Many customers use specific terms and phrases in the cluster description, e. g. "Production", "test", "testing", "ignore alerts", "prd", "prod", "nonprod".
+- Some customers use labels on the `clusters.cluster.x-k8s.io` resource (the cluster's main resource) indicating the importance, e.g:
+    - `odp/environment: development`
+    - `odp/environment: staging`
+    - `odp/environment: production`.
+
+## Proposal
+
+We introduce a label and a schema for the label values to indicate the relative importance of a cluster.
+
+Label name: `giantswarm.io/service-priority`
+
+The label is to be used on the main resource defining the cluster, usually of resource type `clusters.cluster.x-k8s.io`.
+
+To simplify maintenance and avoid logical conflicts, this label SHOULD NOT be applied on other resources of the cluster. Systems MUST ignore the label used on other resources in the same cluster. This means that it is not possible, for example, to differentiate between the importance of several node pools within the same cluster.
+
+### Well-defined label values
+
+There are three well-defined label values to be used.
+
+- `giantswarm.io/service-priority: critical`: This is the highest priority class. Clusters with this label value are considered the most important ones, relative to other clusters not carrying this label. We expect customers to use this class for clusters serving user-facing applications, production traffic, etc.
+
+- `giantswarm.io/service-priority: important`: Clusters carrying this label are considered less important than `critical`, but still more important than clusters classified as `unimportant`. Typical use cases would be staging clusters, clusters handling batch workloads, or development clusters that cannot be replaced easily by creating a new cluster.
+
+- `giantswarm.io/service-priority: unimportant`: The lowest priority class. Customers are expected to use this class for clusters that isn't relied on, or which can be replaced easily and quickly.
```

Comment:
Instead of `unimportant` I would name this level `standard`.

Reply:
I think it has a better feeling to it and contains no negative message to users for example.

Reply:
Well, except for `standard` sounds a lot like this would be the default, which it isn't. `critical` would be default.

Can anyone think of a better term?

Reply:
Yeah, our standard is to treat everything as critical. Maybe something like “inessential”?

Reply:
why not go with the typical priority scale of, high, medium, low (critical could be more than just high)? Unimportant or inessential have connotations that a customer could interpret as service level (e.g. no alerting), but alerting and such is handled separately, so priority is solely on a comparison between two or more clusters.

Reply:
Then here is my alternative suggestion:

- `highest`
- `medium`
- `lowest`

This should make it clear for each of the values, simply from looking at it, what it means relative to others. You don't even have to memorize the entire scale.